aea/cli/check_packages.py

```diff
@@ -324,22 +347,261 @@ def check_public_id(configuration_file: Path) -> None:
+        re.MULTILINE,
+    )
+    if len(matches) == 1:
+        # process the result
+        _, match1, match2 = matches[0]
+        match = match1 if match1 != "" else match2
+        if str(expected_public_id) != match:
+            raise WrongPublicIdError(package_type, expected_public_id, match)
+        return
+
+    # check second pattern of public id: PublicId('...', '...', '...')
+    matches = re.findall(
+        r"^PUBLIC_ID = PublicId\( *['\"](.*)['\"] *, *['\"](.*)['\"] *, *['\"](.*)['\"] *\)$",
+        content,
+        re.MULTILINE,
+    )
+    if len(matches) == 1:
+        # process the result
+        author, name, version = matches[0]
+        actual_public_id_str = f"{author}/{name}:{version}"
+        if str(expected_public_id) != actual_public_id_str:
+            raise WrongPublicIdError(
+                package_type, expected_public_id, actual_public_id_str
+            )
+        return
 
     public_id_code = matches[0]
     if str(expected_public_id) not in public_id_code:
         raise WrongPublicIdError(package_type, expected_public_id, public_id_code)
 
 
+class DependenciesTool:
+    """Tool to work with setup.py dependencies."""
+
+    @staticmethod
+    def get_package_files(package_name: str) -> List[Path]:
+        """Get package files list."""
+        packages_info = list(search_packages_info([package_name]))
+        if len(packages_info) == 0:
+            raise Exception(f"package {package_name} not found")
+        if isinstance(packages_info[0], dict):
+            files = packages_info[0]["files"]
+            location = packages_info[0]["location"]
+        else:
+            files = packages_info[0].files  # type: ignore
+            location = packages_info[0].location  # type: ignore
+        return [Path(location) / i for i in files]  # type: ignore
+
+    @staticmethod
+    def clean_dependency_name(dependecy_specification: str) -> str:
+        """Get dependency name from dependency specification."""
+        match = DEP_NAME_RE.match(dependecy_specification)
+        if not match:
+            raise ValueError(f"Bad dependency specification: {dependecy_specification}")
+        return match.groups()[0]
+
+
+class ImportsTool:
+    """Tool to work with 3rd part imports in source code."""
+
+    @staticmethod
+    def get_imports_for_file(pyfile: Union[str, Path]) -> List[str]:
+        """Get all imported modules for python source file."""
+        with open(pyfile, "r") as f:
+            statements = f.read()
+        instructions = dis.get_instructions(statements)  # type: ignore
+        imports = []
+        is_try_except_block = False
+        for im in instructions:
+            if "SETUP_FINALLY" in im.opname:
+                is_try_except_block = True
+
+            if "POP_EXCEPT" in im.opname:
+                is_try_except_block = False
+
+            if is_try_except_block:
+                continue
+
+            if "IMPORT" in im.opname:
+                imports.append(im)
```

Comment:
this could use some documentation / comment. If we have a `try` (`TRY_FINALLY`) and then import, we consider this optional and do not append to `imports`. `if IMPORT in im.opname` covers `IMPORT_FROM` and `IMPORT_NAME` and `IMPORT_STAR`, not sure how the latter would be dealt with but I believe we do not allow it in the first place.

last two could be merged into a single clause

```
            if "SETUP_FINALLY" in im.opname:
                is_try_except_block = True

            if "POP_EXCEPT" in im.opname:
                is_try_except_block = False

            if is_try_except_block:
                continue

            if "IMPORT" in im.opname:
                imports.append(im)